runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/SparkMetricsContainerStepMap.java

```diff
@@ -27,7 +27,7 @@
 
   @Override
   public String toString() {
-    return new SparkBeamMetric().renderAll().toString();
+    return asAttemptedOnlyMetricResults(this).toString();
```

Comment:
We ran our application built with Beam `2.18.0` and the metrics are displayed correctly, with this fix Spark displays the same results.

The change that broke things up for Spark was part of BEAM-9600 - #11369 .

Reply:
@ibzib fyi

Reply:
Thanks for the heads up. My assumption in #11369 was that the result was meant to be consumed by code, not humans. After this change, metrics would remain not too readable for Flink. Ideally we could find a way to make the Flink metrics protos available to the Python client without `toString`, since `toString` should generally should be human-readable. The problem is that without a job server, there's nowhere to access job information like metrics outside of what Flink provides natively.

Reply:
@ibzib Thanks for details. I agree that we should not rely on `toString()` when it's going to be used by other program. 
I'm sorry but I'm not sure that I fully understand your last sentence - does it seem that it's only specific Flink-related issue?
Regarding readable Flink metrics - would it be better to fix it on FlinkRunner side, like we do with `SparkMetricsContainerStepMap` for Spark Runner?

Reply:
There are two ways of running Beam Python Flink/Spark. One way involves starting a Java job server, the other (newer) way does not require a Java job server and instead uses only Python. When using a Java job server, it's easy to get metrics, since the job server has access to the Flink/Spark context object. But in Python, we rely on the Flink REST API to get metrics, so the formatting of the results displayed there is important.

The reason metrics formatting isn't a problem for Spark is that spark_uber_jar_job_server.py just doesn't implement get_metrics yet.

The problem is that if we want to make `MetricsContainerStepMap::toString` human-readable, we'll need to make the protobuf-formatted metrics accessible somewhere else.

Perhaps there is a compromise though if we can somehow format metrics so they are readable both by the proto parser and humans.

Reply:
@ibzib Thanks for more details. I see the picture bit still not convinced why `MetricsContainerStepMap::toString ` is used for REST API (if I get it right) - dedicated way  to make the protobuf-formatted metrics accessible, as you said, should be better imho. 

Are you ok to merge this fix for Spark metrics?

Reply:
> @ibzib Thanks for more details. I see the picture bit still not convinced why MetricsContainerStepMap::toString is used for REST API (if I get it right) - dedicated way to make the protobuf-formatted metrics accessible, as you said, should be better imho.

The reason I used `toString` is because the Flink REST API displays accumulator values using `toString`. I simply wasn't able to come up with a dedicated way to make the protobuf-formatted metrics accessible.

> Are you ok to merge this fix for Spark Runner metrics?

Yes, this is fine for now. I might need to give Flink runner metrics another look to make sure we don't have a similar problem there.